pkg/githubutils/githubutils.go

```diff
@@ -0,0 +1,74 @@
+package githubutils
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"os"
+
+	"github.com/grafana/plugin-validator/pkg/logme"
+)
+
+func FetchLatestGrafanaSdkRelease() (Release, error) {
+	return FetchGrafanaSdkReleaseByTag("latest")
+}
+
+func FetchGrafanaSdkReleaseByTag(tag string) (Release, error) {
+	if tag == "" {
+		return Release{}, fmt.Errorf("tag is required")
+	}
+
+	apiUrl := "https://api.github.com/repos/grafana/grafana-plugin-sdk-go/releases"
+
+	if tag == "latest" {
+		apiUrl = apiUrl + "/latest"
+	} else {
+		apiUrl = apiUrl + "/tags/" + tag
+	}
+
+	req, err := http.NewRequest("GET", apiUrl, nil)
+	if err != nil {
+		return Release{}, err
+	}
+
+	token := os.Getenv("GITHUB_TOKEN")
```

Comment:
Suggestion: This is probably okay for now but I would use a:

```go
type Client struct {
    token string
}

func NewClient(token string) *Client {
    return &Client{token: token}
}

func (c *Client) FetchGrafanaSDKReleaseByTag(...) {
    // ...
    // use c.token
    //
}
```

```go
// somewhere in main..
ghClient := github.Client(os.Getenv("GITHUB_TOKEN"))
```

and pass the client around. In general it's discouraged to use environment variables in this global-like way. But probably okay for now, I would also consider switching to a package instead in the future like Andres suggested in case we end up making more complicated things with the GitHub API.

Reply:
good idea. I really want to keep this simple without the need for more and more parameters because if I make this function accept the token I'll need to pass it all the way down from the main function to this one and it'll just make the changes downstream too big